### PR TITLE
Feature/testing it 1,3

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do?
 - Adds Cypress tests for iteration 1 and iteration 3 user stories
 - Tests cover:
   -  the main movie list view --> user should see app title and all movie titles (image testing forthcoming lol)
   - 400 level error and appropriate error message for user (500 level error message test forthcoming!!)
   
## How should it be tested?
 - Fetch my branch and run cypress!

### CODE REVIEW

File: cypress/integration/rancid_tomatillos_spec1.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+describe('App', () => {
+
+  it('should be able to visit app and render the correct elements', () => {
+    cy.visit('http://localhost:3000')
+    const allMovieTitles = movieData.movies.map(movie => movie.title);
+
+    allMovieTitles.forEach(movieTitle => {
+      cy.get('div').contains(movieTitle)
+    });
+
```
Comment (cbmackintosh) [inline cypress/integration/rancid_tomatillos_spec1.js:12]: ^ Very clever. Good way to show that ALL the movie titles are displaying on the home page

File: cypress/integration/rancid_tomatillos_spec1.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  //       method: 'GET',
+  //       url: 'https://rancid-tomatillos.herokuapp.com/api/v2/movies'
+  //     },
+  //     {
+  //       statusCode: 500,
+  //     })
+  //     cy.visit('http://localhost:3000')
+  //     .get('h2').should('contain', 'Oop, that/s no good 😅')
+  // })
+
```
Comment (cbmackintosh) [inline cypress/integration/rancid_tomatillos_spec1.js:39]: So my understanding is we don't need to test for 500 errors, but we should still build in error handling for them? Is that correct?

File: cypress/integration/rancid_tomatillos_spec1.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  it('should show an informative error message when the server is down', () => {
+    cy.intercept({
+        method: 'GET',
+        url: 'https://rancid-tomatillos.herokuapp.com/api/v2/movies'
+      },
+      {
+        statusCode: 422,
+      })
+      cy.visit('http://localhost:3000')
+      .get('h2').should('contain', 'Oh wow. This is embarassing. Try again later? 😅')
```
Comment (cbmackintosh) [inline cypress/integration/rancid_tomatillos_spec1.js:25]: Glad we figured this out together. This test may prevent us from using a beforeEach to simplify the code. Once we have our test files consolidated we can look for any repeating expressions to refactor.

**Review by cbmackintosh — APPROVED** [review]

**Comment by m1073496** [comment]
[Cypress: Iterations 1 & 3](https://trello.com/c/lCkw4IDc/26-cypress-iterations-1-3)

**Pull request merged**